optaplanner-docs/src/main/asciidoc/ConstraintStreams/ConstraintStreams-chapter.adoc

```diff
@@ -574,6 +574,121 @@ the stream.
 ====
 
 
+[[constraintStreamsTesting]]
+== Testing a constraint stream
+
+Constraint streams come with a unit testing harness, called "Constraint Verifier".
+Consider the following constraint stream:
+
+[source,java,options="nowrap"]
+----
+    protected Constraint horizontalConflict(ConstraintFactory factory) {
+        return factory
+                .fromUniquePair(Queen.class, equal(Queen::getRowIndex))
+                .penalize("Horizontal conflict", SimpleScore.ONE);
+    }
+----
+
+This is how a simple unit test for this constraint looks, using the Constraint Verifier API:
+
+[source,java,options="nowrap"]
+----
+    private ConstraintVerifier<NQueensConstraintProvider, NQueens> constraintVerifier
+            = ConstraintVerifier.build(new NQueensConstraintProvider(), NQueens.class, Queen.class);
+
+    @Test
+    public void horizontalConflictWithTwoQueens() {
+        Row row1 = new Row(0);
+        Column column1 = new Column(0);
+        Column column2 = new Column(1);
+        Queen queen1 = new Queen(0, row1, column1);
+        Queen queen2 = new Queen(1, row1, column2);
+        constraintVerifier.verifyThat(NQueensConstraintProvider::horizontalConflict)
+                .given(queen1, queen2)
+                .penalizesBy(1);
+    }
+----
+
+This test ensures that the "Horizontal conflict" constraint gives a penalty of `1` when there are two queens on the
+same row.
+Let's take a look at the test in detail.
```

Comment:
Docs (unlike blogs) should be formal and avoid "Let's".
@emmurphy1 wdyt?

Reply:
Yes. Also, don't use " to denote names. I would capitalize Horizontal Conflict. Or you can just make both l/c as I don't see that this is used elsewhere. Either is fine IMO.

This test ensures that the  horizontal conflict constraint assigns a penalty of `1` when there are two queens on the
same row.  

I would just delete the linking text Let's take a look at the test in detail.